src/caselawclient/Client.py

```diff
@@ -646,6 +649,23 @@ def user_can_view_unpublished_judgments(self, username):
         result = multipart_data.parts[0].text
         return result.lower() == "true"
 
+    def user_has_role(self, username, role):
+        vars = {
+            "user": username,
+            "role": role,
+        }
+        return self._send_to_eval(vars, "user_has_role.xqy")
+
+    @cached
```

Comment:
Does it make more sense to cache user_has_role instead?

Reply:
I don't know - `user_has_privilege` is not cached while `user_can_view_unpublished_judgments` is, so I was mirroring that. Can you let me know your thoughts on the privilege based ones? I think if we're going to be caching  `user_has_role` instead of its implementing method, then we should do it for `user_has_privilege` as well?

Reply:
The mirroring is good and it's worth keeping. I think it does make sense to cache the True/False response, rather than the larger response we get back.